src/backend/api/Fusion.Resources.Domain/Queries/Departments/GetRelevantDeparmentProfile.cs

```diff
@@ -0,0 +1,322 @@
+                    {
+                        break;
+                    }
+                    List<string> Reasons = new List<string>();
+                    if (department.Value.Contains("Resources.ResourceOwner") && !department.Key.Contains("*"))
+                    {
+
+                        Reasons.Add("DelegatedManager");
+
+                    }
+                    else if (department.Value.Contains("Resources.FullControl") && department.Key.Contains("*"))
+                    {
+
+                        List<string> reason = new List<string>();
+                        reason.Add("ParentManager");
+                        var DepChildren = await ResolveChildren(department.Key.Trim('*').TrimEnd());
+
+                        await GetDepartmentChildren(department.Key.Trim('*').TrimEnd(), reason, request.Query);
+
+
+                        Reasons.Add("Manager ( Wildcard) ");
+
+
+                    }
+                    else if (department.Value.Contains("Resources.FullControl") && !department.Key.Contains("*"))
+                    {
+                        Reasons.Add("Write Access");
+
+                    }
+                    else
+                    {
+
+                        Reasons.Add("NOT Manager");
+
+                    }
+
+                    if (Reasons.Count > 0)
+                    {
+                        QueryRelevantDepartmentProfile? DepartmentProfile = await GetDepartmentInformation(department.Key.Trim('*').TrimEnd(), Reasons, request.Query);
+                        await StoreResult(lstDepartments, DepartmentProfile);
+                    }
+
+                }
+
+                return lstDepartments;
+            }
+
+
+            private async Task<bool> GetDepartmentChildren(string dep, List<string> reason, ODataQueryParams query)
+            {
+                if (sapIdFound == true)
+                {
+                    return true;
+                }
+                var DepChildren = await ResolveChildren(dep);
+                if (DepChildren != null)
+                {
+                    // recurse over all children categories and add them to the list
+                    foreach (var child in DepChildren.Children)
+                    {
+                        if (sapIdFound == true)
+                        {
+                            break;
+                        }
+                        QueryRelevantDepartmentProfile? ChildDepartmentProfile = await GetDepartmentInformation(child.DepartmentId, reason, query);
+                        await StoreResult(lstDepartments, ChildDepartmentProfile);
+
+                        await GetDepartmentChildren(child.DepartmentId, reason, query);
+                    }
+                }
+                return true;
+            }
+
+            private static async Task<bool> StoreResult(List<QueryRelevantDepartmentProfile?> lstDepartments, QueryRelevantDepartmentProfile? DepartmentProfile)
```

Comment:
What do you store ?
Seems to be a funky name.

Without further comments.
Please have a look at other queries and commands, and try to adjust usage of names.

Get vs Resolve etc.

Reply:
naming is not tought of , just to get a POC of the code. 
so  restructuring and naming is part of the  clenaup.